packages/core/test/acceptance/component_spec.ts

```diff
@@ -303,57 +303,70 @@ describe('component', () => {
        expect(wrapperEls.length).toBe(2);  // other elements are preserved
      });
 
-  it('should invoke `onDestroy` callbacks of dynamically created component', () => {
-    let wasOnDestroyCalled = false;
-    @Component({
-      selector: '[comp]',
-      template: 'comp content',
-    })
-    class DynamicComponent {
-    }
-
-    @NgModule({
-      declarations: [DynamicComponent],
-      entryComponents: [DynamicComponent],  // needed only for ViewEngine
-    })
-    class TestModule {
-    }
-
-    @Component({
-      selector: 'button',
-      template: '<div id="app-root" #anchor></div>',
-    })
-    class App {
-      @ViewChild('anchor', {read: ViewContainerRef}) anchor!: ViewContainerRef;
-
-      constructor(private cfr: ComponentFactoryResolver, private injector: Injector) {}
-
-      create() {
-        const factory = this.cfr.resolveComponentFactory(DynamicComponent);
-        const componentRef = factory.create(this.injector);
-        componentRef.onDestroy(() => {
-          wasOnDestroyCalled = true;
-        });
-        this.anchor.insert(componentRef.hostView);
-      }
-
-      clear() {
-        this.anchor.clear();
-      }
-    }
-
-    TestBed.configureTestingModule({imports: [TestModule], declarations: [App]});
-    const fixture = TestBed.createComponent(App);
-    fixture.detectChanges();
-
-    // Add ComponentRef to ViewContainerRef instance.
-    fixture.componentInstance.create();
-    // Clear ViewContainerRef to invoke `onDestroy` callbacks on ComponentRef.
-    fixture.componentInstance.clear();
-
-    expect(wasOnDestroyCalled).toBeTrue();
+  describe('with ngDevMode', () => {
+    const _global: {ngDevMode: any} = global as any;
+    let saveNgDevMode!: typeof ngDevMode;
+    beforeEach(() => saveNgDevMode = ngDevMode);
+    afterEach(() => _global.ngDevMode = saveNgDevMode);
+    ['ngDevMode off', 'ngDevMode on'].forEach((mode) => {
```

Comment:
I'd propose to add a comment here, since we don't usually run tests in dev and prod modes.
```suggestion
    // In dev mode we have some additional logic to freeze `TView.cleanup` array
    // (see `storeCleanupWithContext` function).
    // The tests below verify that this action doesn't trigger any change in behaviour
    // for prod mode. See https://github.com/angular/angular/issues/40105.
    ['ngDevMode off', 'ngDevMode on'].forEach((mode) => {
```

Reply:
adde, thank you.